src/pages/index.js

```diff
@@ -21,79 +21,187 @@ import {
+  })
+  .catch((err) => console.error(err));
+
 const cardImageModal = new PopupWithImage("#preview-image-modal");
 cardImageModal.setEventListeners();
 
 function handleImageClick(cardData) {
   cardImageModal.open(cardData);
 }
 
-const addCardPopup = new PopupWithForm(
-  "#add-card-modal",
-  handleAddCardFormSubmit
+const deleteModalConfirmation = new PopupConfirmation(
+  "#modal-confirm",
+  async (cardId, cardElement) => {
+    try {
+      await api.deleteCard(cardId);
+      cardElement.remove();
+    } catch (err) {
+      console.error(`Error On Card Deletion ${err}`);
+    }
+  }
 );
-addCardPopup.setEventListeners();
+
+function handleDeleteCard(card) {
+  deleteModalConfirmation.setSubmitAction(() => {
+    api
+      .deleteCard(card.getId())
+      .then(() => {
+        card.remove();
+        deleteModalConfirmation.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  });
+  deleteModalConfirmation.open();
+}
+
+function handleLikeClick(card) {
+  if (card.setIsLiked) {
+    api
+      .dislikeCard(card._id)
+      .then(() => {
+        card.handleLikeIcon(); // here
+      })
+      .catch(console.error);
+  } else {
+    api
+      .likeCard(card._id)
+      .then(() => {})
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+}
 
 const editProfilePopup = new PopupWithForm(
   "#profile-edit-modal",
-  handleProfileEditSubmit
+  async (data) => {
+    await api.updateUserProfile({
+      name: data.title,
+      about: data.description,
+    });
+    userInfo.setUserInfo({
+      title: data.title,
+      description: data.description,
+    });
+  }
 );
 editProfilePopup.setEventListeners();
 
-function handleProfileEditSubmit(formData) {
-  userInfo.setUserInfo({
-    name: formData.title,
-    description: formData.description,
-  });
-  editProfilePopup.close();
-}
+const cardSection = new Section(
+  {
+    items: initialCards,
+    renderer: (item) => {
+      const card = createCard(item);
+      cardSection.addItem(card);
+    },
+  },
+  ".cards__list"
+);
 
-function handleAddCardFormSubmit(inputValues) {
-  console.log(inputValues);
-  const name = inputValues.title;
-  const link = inputValues.Url;
-  const cardData = { name, link };
-  cardListEl.addItem(createCard(cardData));
-  console.log(cardData);
-  addCardPopup.close();
-  addCardFormElement.reset();
-}
+const addCardPopup = new PopupWithForm("#add-card-modal", async (formData) => {
+  const name = formData.title;
+  const link = formData.Url;
+
+  const res = await api.createCard({ name, link });
+  cardSection.addItem(createCard(res));
+});
+
+addCardPopup.setEventListeners();
+
+const editAvatarModal = new PopupWithForm(
+  "#edit-avatar-modal",
+  async (formData) => {
+    const avatarUrl = formData.avatar;
+    const userData = await api.updateAvatar(avatarUrl);
+    userInfo.setUserAvatar(userData.avatar);
+  }
+);
+
+editAvatarButton.addEventListener("click", () => {
+  editAvatarModal.open();
```

Comment:
Please, call `resetValidation` to disable the button on opening if there is no other ways of disabling it after submission